lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -85,6 +86,133 @@ const char DEV_NULL[] = "/dev/null";
 
 namespace lldb_dap {
 
+DAPSessionManager &DAPSessionManager::GetInstance() {
+  // NOTE: intentional leak to avoid issues with C++ destructor chain
+  static DAPSessionManager *instance = nullptr;
+  if (!instance) {
+    instance = new DAPSessionManager();
+  }
+  return *instance;
```

Comment:
Is this thread safe? I think you need mutex protection or `std::call_once` for it (or Double-checked locking with manual mutex lock). 
Since C++11, static local variable initialization is guaranteed to be thread-safe but since you are using pointer here, I do not think C++ compiler is providing you thread safe guarantee.